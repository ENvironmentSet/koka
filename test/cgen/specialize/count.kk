fun count(start : int, end : int) : console ()
  if start == end then
    return ()
  
  start.println
  count(unsafe-decreasing(start - 1), end)

fun main()
  count(10, 0)