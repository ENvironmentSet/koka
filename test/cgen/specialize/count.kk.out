10
9
8
7
6
5
4
3
2
1
 
cgen/specialize/count/.lift000-main: (start : int) -> console ()
cgen/specialize/count/count: (start : int, end : int) -> console ()
cgen/specialize/count/main: () -> console ()